### PR TITLE
Add MOBMOD_NO_WIDESCAN & CMobEntity::isWideScannable()

### PR DESCRIPTION
**Pull request by hooksta4**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
Adds ability to remove mobs from widescan. 

## Steps to test these changes
mob:setMobMod(xi.mobMod.MOBMOD_NO_WIDESCAN, 1)

### CODE REVIEW

File: scripts/globals/status.lua [hunk 1]
```diff
@@ -2587,6 +2587,7 @@ xi.mobMod =
     TARGET_DISTANCE_OFFSET = 73, -- Adjusts how close a mob will move to it's target. 12 = 1.2 yalm. Positive values to go closer, negative farther.
     ONE_WAY_LINKING        = 74, -- Will link with other mobs in its party (typically the same mob family) while roaming, but will not let others link with it once engaged
     CAN_PARRY              = 75, -- Check if a mob is allowed to have parry rank (Rank Value 1-5)
+    MOBMOD_NO_WIDESCAN     = 76, -- Disables widescan for a specific mod
```
Comment (Frankie-hz) [inline scripts/globals/status.lua]: I would change the name to NO_WIDESCAN to match the rest of the mobmods. Only needs to be in status.lua

**Review by Frankie-hz — APPROVED** [review]

File: scripts/globals/status.lua (hunk 1 above)
Comment (claywar) [reply scripts/globals/status.lua]: Agreed on the above comment.  We know this is a mobmod due to the enum declaration.

Comment (hooksta4) [reply scripts/globals/status.lua]: Very well. Thank you.

**Review by Frankie-hz — APPROVED** [review]

**Review by WinterSolstice8 — APPROVED** [review]

File: scripts/globals/status.lua [hunk 2]
```diff
@@ -2587,6 +2587,7 @@ xi.mobMod =
     TARGET_DISTANCE_OFFSET = 73, -- Adjusts how close a mob will move to it's target. 12 = 1.2 yalm. Positive values to go closer, negative farther.
     ONE_WAY_LINKING        = 74, -- Will link with other mobs in its party (typically the same mob family) while roaming, but will not let others link with it once engaged
     CAN_PARRY              = 75, -- Check if a mob is allowed to have parry rank (Rank Value 1-5)
+    NO_WIDESCAN            = 76, -- Disables widescan for a specific mod
```
Comment (zach2good) [inline scripts/globals/status.lua]: specific mob*

File: src/map/mob_modifier.h [hunk 3]
```diff
@@ -104,6 +104,7 @@ enum MOBMODIFIER : int
     MOBMOD_TARGET_DISTANCE_OFFSET = 73, // Adjusts how close a mob will move to it's target. 12 = 1.2 yalm. Positive values to go closer, negative farther.
     MOBMOD_ONE_WAY_LINKING        = 74, // Will link with other mobs in its party (typically the same mob family) while roaming, but will not let others link with it once engaged
     MOBMOD_CAN_PARRY              = 75, // Check if a mob is allowed to have parry rank (Rank Value 1-5)
+    MOBMOD_NO_WIDESCAN            = 76, // Disables widescan for a specific mod
```
Comment (zach2good) [inline src/map/mob_modifier.h]: specific mob*

**Review by zach2good — CHANGES_REQUESTED** [review]

File: src/map/mob_modifier.h (hunk 3 above)
Comment (hooksta4) [reply src/map/mob_modifier.h]: i thought i had made those changes... opps.

**Review by Xaver-DaRed — APPROVED** [review]

**Review by zach2good — APPROVED** [review]

**Pull request merged**